agent/testflinger_agent/agent.py

```diff
@@ -111,6 +124,50 @@ def mark_device_offline(self):
         # Create the offline file, this should work even if it exists
         open(self.get_offline_files()[0], "w").close()
 
+    def unpack_attachments(self, job_data: dict, cwd: Path):
+        """Download and unpack the attachments associated with a job"""
+
+        # download attachment archive to a unique temporary folder
+        archive_dir = tmp_dir()
+        archive_path = self.client.get_attachments(
+            job_id=job_data["job_id"], path=archive_dir
+        )
+        if archive_path is None:
+            raise FileNotFoundError
```

Comment:
Could you please also add any logging or exception message here to include the path to the file that is not found? I guess it would be helpful for a user to see what caused the failure

Reply:
You're right, I will add a message to the exception. Logging is taken care of by the `except` block that catches the exception (and will use that message)